src/BoilerPlate/Game.cpp

```diff
@@ -0,0 +1,253 @@
+		break;
+	case SDL_SCANCODE_F:
+		ToggleDebuggingMode();
+		break;
+	case SDL_SCANCODE_E:
+		IncreaseAsteroids();
+		break;
+	case SDL_SCANCODE_Q:
+		DecreaseAsteroids();
+		break;
+	case SDL_SCANCODE_R:
+		RespawnShip();
+		break;
+	case SDL_SCANCODE_Z:
+		ResetGame();
+	}
+
+}
+
+void Game::OnKeyUp(SDL_KeyboardEvent keyBoardEvent)
+{
+	switch (keyBoardEvent.keysym.scancode)
+	{
+	case SDL_SCANCODE_W:
+		ship.SetMovingForward(false);
+		break;
+	}
+}
+
+void Game::Update(float deltaTime, float currentHeight, float currentWidth)
+{
+	ship.Update(deltaTime);
+	ship.UpdateFrameSize(currentHeight, currentWidth);
+
+	for (int i = 0; i < asteroids.size(); i++)
+	{
+		asteroids[i].Update(deltaTime);
+	}
+
+	for (int i = 0; i < asteroids.size(); i++)
+	{
+		asteroids[i].UpdateFrameSize(currentHeight, currentWidth);
+	}
+
+
+	UpdateCollision();
+}
+
+
+void Game::Render()
+{
+	glClearColor(colors.slateBlue.redValue, colors.slateBlue.greenValue,
+		colors.slateBlue.blueValue, colors.slateBlue.alphaValue);
+	glClear(GL_COLOR_BUFFER_BIT);
+
+
+	ship.Render();
+
+
+	for (int i = 0; i < asteroids.size(); i++)
+	{
+		asteroids[i].Render();
+	}
+
+	if (debuggingMode)
+		DrawDebugCollisionLines();
+
+}
+
+void Game::UpdateFrameSize(float currentHeight, float currentWidth)
+{
+	height = currentHeight;
+	width = currentWidth;
+}
+
+void Game::ToggleDebuggingMode()
+{
+	ship.Respawn();
+
+	if (debuggingMode == false)
+	{
+		debuggingMode = true;
+		ToggleEntityDebug();
+	}
+
+	else
+	{
+		debuggingMode = false;
+		ToggleEntityDebug();
+		playerLife = 3;
+	}
+
+
+
+	if (debuggingMode)
+	{
+		ship.SetDebuggingMode(true);
+	}
+
+}
+
+void Game::ToggleEntityDebug()
+{
+	ship.ToggleDebuggingMode();
+
+	for (int i = 0; i < asteroids.size(); i++)
+	{
+		asteroids[i].ToggleDebuggingMode();
+	}
+}
+
+
+void Game::IncreaseAsteroids()
+{
+	if (debuggingMode)
+	{
+		asteroids.push_back(Asteroid());
+	}
+
+	if (debuggingMode)
+	{
+		asteroids[asteroids.size() - 1].SetDebuggingMode(true);
+	}
+
+}
+
+void Game::DecreaseAsteroids()
+{
+	if (debuggingMode)
+	{
+		if (asteroids.size() != 0)
+			asteroids.pop_back();
+	}
+
+}
+
+void Game::DrawDebugCollisionLines()
+{
+
+	glLoadIdentity();
+
+	glBegin(GL_LINE_LOOP);
+	glColor3f(0.863, 0.078f, 0.235f);
+
+	for (int i = 0; i < asteroids.size(); i++)
+	{
+
+
+		if (CalculateDistance(ship, asteroids[i]) <= ship.GetHitRadius() * 2 + asteroids[i].GetHitRadius())
+		{
+			glVertex2f(ship.GetPosition().x, ship.GetPosition().y);
+			glVertex2f(asteroids[i].GetPosition().x, asteroids[i].GetPosition().y);
+		}
+	}
+	glEnd();
+
+}
+
+float Game::CalculateDistance(Entity entity1, Entity entity2)
+{
+	float distance = sqrtf(((entity2.GetPosition().x - entity1.GetPosition().x)*(entity2.GetPosition().x - entity1.GetPosition().x)) +
+		((entity2.GetPosition().y - entity1.GetPosition().y)*(entity2.GetPosition().y - entity1.GetPosition().y)));
+
+	return distance;
+}
+
+bool Game::DetectCollision(Entity entity1, Entity entity2)
```

Comment:
I'd avoid using number suffixes for variables, there should be a better naming approach for this, try to think about it ;-)